### PR TITLE
WASM Support for Blind Signatures

### PR DESCRIPTION
**Pull request by gakonst**
Opening this for initial feedback. 

1. Stages Cargo.lock to make sure our code doesn't break as our dependencies evolve
2. Updates bls-crypto to latest, so that we can use the new hash to curve algos which we wrote
3. Adds WASM functionalities & examples. The user provides a seed and a scalar is generating from that seed. Would've been nicer if we could directly pass the RNG so that this generates a different scalar each time

### CODE REVIEW

File: examples/tblind.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+const polynomial = keys.polynomial
+
+// each of these shares proceed to sign teh blinded sig
+let sigs = []
+for (let i = 0 ; i < keys.num_shares(); i++ ) {
+    const sig = threshold.partial_sign(keys.get_share(i), blind_msg)
+    sigs.push(sig)
+}
+
+// The combiner will verify all the individual partial signatures...
```
Comment (gakonst) [inline examples/tblind.js:29]: This part is commented out as it's blocked by https://github.com/celo-org/celo-threshold-bls-rs/pull/9

Comment (kobigurk) [reply examples/tblind.js:29]: Fixed by https://github.com/celo-org/celo-threshold-bls-rs/pull/9/commits/857465099ef2c06f83a879b8f8dd4bacf8e0fb6e. Partial signing and verification should not map to curve.

File: src/sig/wasm.rs [hunk 2]
```diff
@@ -0,0 +1,300 @@
+use wasm_bindgen::prelude::*;
+
+use rand_chacha::ChaChaRng;
+use rand_core::{RngCore, SeedableRng};
+
+use crate::{
+    curve::zexe::PairingCurve as Bls12_377,
```
Comment (nikkolasg) [inline src/sig/wasm.rs:7]: I would really much prefer to keep the signatures algorithm, the curves they use and some specific compiling (wasm) separated as much as possible. Here it introduces both wasm and zexe curve in the sig package.
Would it be possible to make a `threshold::wasm` package ? that can also compiles using other curves ? Like can it be made with the generic traits `Curve` `Point` `Scalar` ?

Comment (gakonst) [reply src/sig/wasm.rs:7]: Wasm support is not really generic friendly so we'd have to concretely instantiate it for each curve. I will move it to a separate module though, np.

File: src/wasm.rs [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    let scalar = unsafe { &*scalar };
+
+    BlindThresholdSigs::unblind(&scalar, &blinded_msg)
+        .map_err(|_| JsValue::from_str("could not unblind signature"))
+}
+
+#[wasm_bindgen]
+/// Verifies the signature after it has been unblinded. Users will call this on the
+/// threshold signature against the full public key
+pub fn verify_sign(public_key: *const PublicKey, msg: Vec<u8>, signature: Vec<u8>) -> Result<bool> {
```
Comment (kobigurk) [inline src/wasm.rs:65]: How about we return an error in case of bad verification? IMO APIs that return bool are easy to ignore.

**Pull request merged**